model/postgres/postgres_server.rb

```diff
@@ -95,14 +98,22 @@ def check_pulse(session:, previous_pulse:)
     session[:db_connection] ||= Sequel.connect(adapter: "postgres", host: health_monitor_socket_path, user: "postgres")
 
     reading = begin
-      session[:db_connection]["SELECT 1"].all && "up"
+      lsn_function = primary? ? "pg_current_wal_lsn()" : "pg_last_wal_receive_lsn()"
+      lsn_monitor.last_known_lsn = session[:db_connection]["SELECT #{lsn_function} AS lsn"].first[:lsn]
```

Comment:
This is an unlogged table. Loosing updates are one thing but if the database is crashed, the whole data is truncated. When we restore respirate, `lsn_monitor` model is gone, too. Wouldn't this line start failing in that case since it's `nil`? We only create the entity in the `assemble`.

Reply:
Or after failover, will this relationship be still in tact? I don't think so, we need a way to recreate this entity.

Reply:
good catch. will fix this case.

Reply:
Fixed it with insert conflict.